milli/src/search/new/ranking_rule_graph/proximity/compute_docids.rs

```diff
@@ -76,10 +74,10 @@ pub fn compute_docids(
         if right_derivs.len() > 1 {
             let universe = &universe;
             if let Some(left_phrase) = left_phrase {
-                if universe.is_disjoint(ctx.get_phrase_docids(left_phrase)?) {
+                if universe.is_disjoint(ctx.get_phrase_docids(None, left_phrase)?) {
                     continue;
                 }
-            } else if let Some(left_word_docids) = ctx.word_docids(left_word)? {
+            } else if let Some(left_word_docids) = ctx.word_docids(Some(universe), left_word)? {
```

Comment:
It is unclear that we want to pass Some(universe) here, especially since we check `is_disjoint` right afterwards

Reply:
would be clearer with a is_empty